ReactiveCocoa/Objective-C/RACSignal.m

```diff
@@ -107,31 +107,36 @@ - (RACSignal *)bind:(RACStreamBindBlock (^)(void))block {
 	return [[RACSignal createSignal:^(id<RACSubscriber> subscriber) {
 		RACStreamBindBlock bindingBlock = block();
 
-		NSMutableArray *signals = [NSMutableArray arrayWithObject:self];
+		__block NSUInteger signalsCount = 1;   // indicates self
```

Comment:
It looks to me like `OSAtomicDecrement32Barrier` would be safe. Once `signalsCount` is `0`, it will never be incremented again. So `sendCompleted` does not need to be called while holding the lock.